curs_lib.c

```diff
@@ -792,6 +792,15 @@ void mutt_flushinp(void)
   flushinp();
 }
 
+void mutt_flush_stdin(void)
+{
+  int c = fgetc(stdin);
+  while (c != '\n' && c != EOF)
+  {
+    c = fgetc(stdin);
+  }
+}
```

Comment:
How about using a `do ... while()` loop to reduce the duplication:
```c
int c;
do    
{ 
  c = fgetc(stdin);
} while ((c != '\n') && (c != EOF));
```

Reply:
I do not like that form (actually mine neither), but there is a clear advantage, will change it.